### ISSUE TITLE
non-solid matte

### ISSUE THREAD

**Issue by incanus** [mapbox/mapbox-gl-native#119]
We should consider a non-solid matte for better context. Consider Apple Maps's grid: 

https://dl.dropboxusercontent.com/u/575564/matte.mov

![matte](https://f.cloud.github.com/assets/17722/2520156/93393a78-b482-11e3-9109-be581d8a0efc.png)

**Comment by soupman99** [comment]
+1

**Comment by 1ec5** [comment]
This feature is important because, especially in a full-screen-map application on iOS, the user would perceive a hang between the time the style is set and the time content starts to appear. If they’re in the midst of a gesture, the effect can be especially jarring because the user may end up panning or zooming farther than they expected to.

In order for this feature to match the user experience provided by MapKit, the grid pattern would have to be independent of any style. With respect to any given tile coordinate, it would have to appear from the moment the style is set (before the style is downloaded or parsed) to the moment any non-background layer appears at that tile coordinate. Either this pattern would appear above any background layer, or the background layer would automatically remain hidden (exposing the pattern beneath it) until a non-background layer can appear.

**Comment by riastrad** [comment]
Sorry @stale, but this is still a sometimes requested feature - so I think it's worth revisiting whether or not this would be possible to include now that some time has passed.

**Comment by ulusoyca** [comment]
This issue was referenced in https://github.com/mapbox/mapbox-gl-native/issues/11814 
I agree with @riastrad that we need this feature to distinguish background layer and unloaded tile visually

**Comment by tobrun** [comment]
GL-native supports the same UX as GL-JS shown in https://github.com/mapbox/mapbox-gl-native/issues/119#issuecomment-44676773. This is a better UX as showing a grid or nothing at all. @ulusoyca could you elaborate why this is needed for your use-case?

**Comment by ulusoyca** [comment]
@tobrun Correct me if I am wrong but the comment you pointed assumes that there are preloaded tiles. In my use case, there is no guarantee of preloaded tiles so it makes difficult to distinguish unloaded tile from background layer

**Comment by tobrun** [comment]
that is correct, you will just see the background color. One way to achieve your effect, haven't verified this but should work, would be to updating your style with:
 - adding a fill layer above background layer with current background color and linking a geometry spanning the world.
 - updating background layer to use [background-pattern](https://docs.mapbox.com/mapbox-gl-js/style-spec/#paint-background-background-pattern) with something as a grid shown in OP

**Comment by ulusoyca** [comment]
@tobrun 
>adding a fill layer above background layer with current background color and linking a geometry spanning the world.
1. Any instructions on how I can link a geometry spanning the world?
2. Should I add the fill layer programmatically? Is it doable in mapbox studio?

**Comment by ulusoyca** [comment]
Ok, I was able to add the layer through Mapbox Studio with the following dataset:
```
  "geometry": {
    "coordinates": [
      [
        [-180, -90],
        [180, -90],
        [180, 90],
        [-180, 90],
        [-180, -90]
      ]
    ],
    "type": "Polygon"
  },
  "type": "Feature",
  "properties": {
    "name": "full_world_cover"
  }
```
